bundle/src/main/java/com/adobe/acs/commons/mcp/impl/processes/asset/FileOrRendition.java

```diff
@@ -0,0 +1,158 @@
+/*
+ * #%L
+ * ACS AEM Commons Bundle
+ * %%
+ * Copyright (C) 2017 Adobe
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package com.adobe.acs.commons.mcp.impl.processes.asset;
+
+import com.adobe.acs.commons.mcp.impl.processes.asset.AssetIngestor.HierarchialElement;
+import com.adobe.acs.commons.mcp.impl.processes.asset.AssetIngestor.Source;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.net.URLConnection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.TreeMap;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Abstraction of a file which might be an asset or a rendition of another asset
+ */
+public class FileOrRendition extends HashMap<String, String> implements HierarchialElement {
```

Comment:
This isn't really critical since this class isn't API, but it would be better to *not* extend HashMap here. See Effective Java, Item #16